cmd/gen-crl/main.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"path"
+	"time"
+
+	"github.com/jmhodges/clock"
+	"github.com/letsencrypt/boulder/pkcs11helpers"
+)
+
+type crlDef struct {
+	num   int
+	start time.Time
+	end   time.Time
+}
+
+func generateDefs(clk clock.Clock, total, initialNum int, startStr string, period, overlap time.Duration) ([]crlDef, error) {
+	if overlap >= period {
+		return nil, errors.New("CRL overlap must be smaller than the validity period")
+	}
+	var crlDefs []crlDef
+	start := clk.Now()
+	if startStr != "" {
+		var err error
+		start, err = time.Parse(time.RFC3339, startStr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to parse start-date %q: %s", startStr, err)
```

Comment:
It seems like this validation logic should be done in `main` where the other flags are validated; then this function can take a `time.Time` as a parameter instead of string.